test/integration/gc_test.ts

```diff
@@ -114,6 +114,92 @@ describe('Garbage Collection', function () {
     assert.equal(root, clone);
   });
 
+  it('garbage collection test4', async function ({ task }) {
+    type TestDoc = { point?: { x?: number } };
+    const docKey = toDocKey(`${task.name}-${new Date().getTime()}`);
+    const doc1 = new yorkie.Document<TestDoc>(docKey);
+    const doc2 = new yorkie.Document<TestDoc>(docKey);
+
+    const client1 = new yorkie.Client(testRPCAddr);
+    const client2 = new yorkie.Client(testRPCAddr);
+
+    await client1.activate();
+    await client2.activate();
+
+    await client1.attach(doc1, { isRealtimeSync: false });
+    doc1.update((root) => (root.point = { x: 0 }));
+    await client1.sync();
+    await client2.attach(doc2, { isRealtimeSync: false });
+
+    doc1.update((root) => {
+      delete root.point;
+    });
+    assert.equal(doc1.getGarbageLen(), 2);
+
+    doc2.update((root) => {
+      delete root.point?.x;
+    });
+    assert.equal(doc2.getGarbageLen(), 1);
+
+    await client1.sync();
+    await client2.sync();
+    await client1.sync();
+
+    assert.equal(doc1.getGarbageLen(), 2);
+    assert.equal(doc2.getGarbageLen(), 2);
+
+    assert.equal(doc1.garbageCollect(MaxTimeTicket), 2);
+    assert.equal(doc2.garbageCollect(MaxTimeTicket), 2);
+
+    await client1.deactivate();
+    await client2.deactivate();
+  });
+
+  it('garbage collection test5', async function ({ task }) {
+    type TestDoc = { point?: { x?: number; y?: number } };
+    const docKey = toDocKey(`${task.name}-${new Date().getTime()}`);
+    const doc1 = new yorkie.Document<TestDoc>(docKey);
+    const doc2 = new yorkie.Document<TestDoc>(docKey);
+
+    const client1 = new yorkie.Client(testRPCAddr);
+    const client2 = new yorkie.Client(testRPCAddr);
+
+    await client1.activate();
+    await client2.activate();
+
+    // 1. initial state
+    await client1.attach(doc1, { isRealtimeSync: false });
+    doc1.update((root) => (root.point = { x: 0, y: 0 }));
+    await client1.sync();
+    await client2.attach(doc2, { isRealtimeSync: false });
+
+    // 2. client1 updates doc
+    doc1.update((root) => {
+      delete root.point;
+    });
+    assert.equal(doc1.getGarbageLen(), 3); // point, x, y
+
+    // 3. client2 updates doc
+    doc2.update((root) => {
+      delete root.point?.x;
+    });
+    assert.equal(doc2.getGarbageLen(), 1); // x
+
+    await client1.sync();
+    await client2.sync();
+    await client1.sync();
+
+    assert.equal(doc1.getGarbageLen(), 3); // point, x, y, x
+    assert.equal(doc2.getGarbageLen(), 3); // x, point, x, y
+
+    // Actual garbage-collected nodes
+    assert.equal(doc1.garbageCollect(MaxTimeTicket), 3); // point, x, y
+    assert.equal(doc2.garbageCollect(MaxTimeTicket), 3); // point, x, y
```

Comment:
OK, I'll apply your comment.
 

I have a question about your comment.

> It would be a good to add code to verify if getGarbageLen() and doc.garbageCollect(MaxTimeTicket) return the same values to ensure the test's purpose.

Does it mean adding the similar code below in `gc_test.ts` file and replacing duplicated code?

```Typescript
const assertGarbageCollect = (doc: Doc, expect: number) => {
    // Check whether getGarbageLen and garbageCollect return exepct number. 
    assert.equal(...);
};
```

Reply:
I thought of simply verifying that the numbers match:
```ts
const gcNodeCount = 2;
assert.equal(doc1.getGarbageLen(), gcNodeCount);
assert.equal(doc2.getGarbageLen(), gcNodeCount);

assert.equal(doc1.garbageCollect(MaxTimeTicket), gcNodeCount);
assert.equal(doc2.garbageCollect(MaxTimeTicket), gcNodeCount);
```